src/validation.cpp

```diff
@@ -1534,6 +1534,7 @@ void UpdateCoinsAndStakes(const CChainParams& params, const CTransaction& tx, CC
         }
     }
     AddCoins(inputs, tx, nHeight, false, fStake, nStakeOutputNumber);
+    stakes.removeOldFreeTxInfos(nHeight);
 }
```

Comment:
TBH the UpdateCoinsAndStakes function looks bad. At least few method extractions are needed.
// check if any stakes are being spent - can be extracted 
// check if any new stakes are being deposited. - can be extracted

Reply:
Good idea, will be done.